src/wast-parser.cc

```diff
@@ -1499,28 +1499,85 @@ Result WastParser::ParsePlainInstr(std::unique_ptr<Expr>* out_expr) {
   return Result::Ok;
 }
 
+// Current Simd const type is V128 const only.
+// The current expected V128 const lists is:
+// i32 0xXXXXXXXX 0xXXXXXXXX 0xXXXXXXXX 0xXXXXXXXX
+Result WastParser::ParseSimdConst(Const* const_, Type in_type, int32_t nSimdConstBytes) {
+  WABT_TRACE(ParseSimdConst);
+
+  // Parse the Simd Consts according to input data type.
+  switch (in_type) {
+    case Type::I32: {
+      const_->loc = GetLocation();
+      int nCount = nSimdConstBytes/sizeof(uint32_t);
+      // Meet expected "i32" token. start parse 4 i32 consts
+      for(int i=0; i<nCount; i++) {
+        Location loc = GetLocation();
+
+        // Expected one 0xXXXXXXXX number
+        if (!PeekMatch(TokenType::Nat))
+          return ErrorExpected({"an Nat literal"}, "123");
+
+        Literal literal = Consume().literal();
+
+        string_view sv = literal.text;
+        const char* s = sv.begin();
+        const char* end = sv.end();
+        Result result;
+
+        result =
+            ParseInt32(s, end, &(const_->v128_bits.v[i]), ParseIntType::SignedAndUnsigned);
+
+        if (Failed(result)) {
+          Error(loc, "invalid literal \"%s\"", literal.text.c_str());
+          return Result::Error;
+        }
+      }
+      break;
+    }
+
+    default:
+      Error(const_->loc, "ParseSimdConst called with invalid input const type, Type::I32 expected");
```

Comment:
Since this will be displayed to the user, it's better to avoid internal function names. Maybe just "expected i32 at start of simd constant". What do you think?

Reply:
Agree.